### PR TITLE
Added kustomize support in docker sandbox

### PR DESCRIPTION
**Pull request by evalsocket**
Context : In flytectl sandbox, I am introducing a new flag i.e. `--kustomize`, But for that container should build manifest before deployment in the entry point.`. Currently, Pytest-flyte maintains the logic for kustomize build-in entry point (https://github.com/flyteorg/pytest-flyte/tree/main/src/pytest_flyte/docker). 

- I used the same implementation that pytest has but after these changes flytectl and pytest can share the same dependency.

### CODE REVIEW

**Comment by EngHabu** [comment]
I don't think pytest *requires* kustomize... it was just whatever was available at the time... Let's hold off this PR until we discuss.

In fact, I would like to understand if there are anything special needed for pytest or whether we can just use sandbox as is... @jeevb @cosmicBboy @wild-endeavor may be able to help clarify that...

**Comment by jeevb** [comment]
The goal was to eventually support setting flyte component versions programmatically in pytest. Could still be useful for that.

@evalsocket Will kubectl apply -k not suffice for this?

**Comment by evalsocket** [comment]
@jeev yes we can use that. I can remove the kustomize install but is it ok if I move the logic from pytest to sandbox ?

**Comment by jeevb** [comment]
> @jeev yes we can use that. I can remove the kustomize install but is it ok if I move the logic from pytest to sandbox ?

Fine by me, but will defer to others.

File: docker/sandbox/kustomization.yaml.tmpl
Comment (jeevb) [inline docker/sandbox/kustomization.yaml.tmpl:8]: Probably unnecessary since this file is not going to be modified directly right? Is the idea to mount a modified file here to override the default sandbox config?

File: docker/sandbox/kustomization.yaml.tmpl
Comment (jeevb) [inline docker/sandbox/kustomization.yaml.tmpl:1]: Probably worth removing the .tmpl suffix in the filename since you are already doing so with the copy.

File: docker/sandbox/Dockerfile
Comment (jeevb) [inline docker/sandbox/Dockerfile:81]: How about copying into /flyteorg/share directly and installing via kubectl apply -k instead of -f?

File: docker/sandbox/kustomization.yaml.tmpl
Comment (evalsocket) [reply docker/sandbox/kustomization.yaml.tmpl:8]: yes, I will mount the user-provided kustomize file and the entry point script will handle the logic. By default, it will not override anything.

File: docker/sandbox/kustomization.yaml
Comment (jeevb) [inline docker/sandbox/kustomization.yaml:1]: Whitespace

**Review by jeevb — APPROVED** [review]
LGTM.

File: docker/sandbox/Dockerfile
Comment (EngHabu) [inline docker/sandbox/Dockerfile:73]: I would actually prefer to move this to install using helm as we discussed... and maybe enable users to provide a values file to be overridden... basically reducing the places we use kustomize and making sure helm consistently work... I think introducing a kustomization.yaml here makes that job harder....

**Review by EngHabu — APPROVED** [review]

**Pull request merged**
